src/core/dataframe.js

```diff
@@ -193,18 +193,23 @@ export default class Dataframe {
         // Linear search for all features
         // Tests triangles instead of polygons since we already have the triangulated form
         // Moreover, with an acceleration structure and triangle testing features can be subdivided easily
-        for (let i = 0; i < vertices.length; i += 6) {
-            if (i >= breakpoints[featureIndex]) {
-                featureIndex++;
-            }
+        let scale;
+        let computeScale = ()=>{
```

Comment:
I think If we extract this to a function with parameters instead inline defining it the code will look cleaner.


```js
 _computeScale(columnNames, featureIndex, vizWidth, widthScale) {
        const f = {};
        columnNames.forEach(name => {
            f[name] = this.properties[name][featureIndex];
        });
        // Line with is saturated at 336px
        const lineWidth = Math.min(vizWidth.eval(f), 336);
        // width is a diameter and scale is radius-like, we need to divide by 2
        return lineWidth / 2 * widthScale;
    }
```

Reply:
You would need to have two `computeScale` functions because the code for lines is different than the code for polygons.

Reply:
is the same it only changes the name of `vizWidth/vizStrokeWidth` parameter. Rename it to `width` and we got a generic one :trollface: